### DISCUSSION TITLE
Pylance not hinting parameters that are passed to parent through **kwargs

### DISCUSSION THREAD

**Discussion by samkanje** [microsoft/pylance-release#4465] [Enhancement]
## Environment data
Pylance language server 2023.5.51 (pyright 7784093a)
OS: MacOS 13.3.1 (a) (22E772610a)
Python3.11 venv

Also tested the release version of Pylance with same results.

This is a followup to #4449 which was closed. The code I provided then was not quite complete, but the code bellow is.

PyCharm correctly infers the parameters being flowed through to the parent __init__() (via **kwargs) while Pylance is not.

Changing the code is also not an option for me because the **kwargs pattern below is being used in an external library that I am using in my project and I have no control over it.

## Code Snippet

```python
class Parent:
    def __init__(self, p_1: str, p_2: str = ""):
        self.p_1 = p_1
        self.p_2 = p_2

    def print_p_1_from_parent(self):
        print(self.p_1)

class Child(Parent):
    def __init__(self, **kwargs):
        super().__init__(**kwargs)
```

## Repro Steps

With the two classes above, parameter hinting while creating an instance of the Child class does not show the parameters defined in the parent.

## Expected behaviour

As observed in PyCharm

![CleanShot 2023-06-03 at 05 04 32](https://github.com/microsoft/pylance-release/assets/8049629/9c97141b-017a-4445-8079-4592f4370549)

## Actual behavior

![CleanShot 2023-06-03 at 05 05 47](https://github.com/microsoft/pylance-release/assets/8049629/4e632883-61ac-4a14-8591-ef9c03e4cc09)

**Comment by heejaechang** [comment]
I think it would be good enhancement to add to our completion provider. thank you for the suggestion!

tagging @luabud

**Comment by debonte** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.

**Comment by xaviergmail** [comment]
This feature would be great. I'm extending classes from AWS CDK to have some sane defaults and I don't know every possible keyword argument that's available to me when using these subclasses. PyCharm seems to do this without a problem. Is there a known workaround for vscode/pycharm? Edit: Jedi seems to do this as well.

**Comment by rb83** [comment]
Just got bitten by this as well. I would think that this pattern is rather often used and should be recognized by pylance

**Comment by mrferguson-md** [comment]
I routinely run into this, would love to see this enhancement make it in.
